Tests/Unit/NetworkServer/CupsUpdateInfoResponseHeaderTests.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit.NetworkServer
+{
+    using System;
+    using System.Buffers;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using LoRaWan.NetworkServer.BasicsStation;
+    using Xunit;
+
+    public class CupsUpdateInfoResponseHeaderTests
+    {
+        private readonly string updateUriString = "https://localhost:1234";
+
+        [Fact]
+        public void Serialize_WithNoUpdates()
+        {
+            var updateResponseHeader = new CupsUpdateInfoResponseHeader
+            {
+                SignatureKeyCrc = 0,
+                UpdateDataLength = 0,
+            };
+
+            using var memoryPool = MemoryPool<byte>.Shared.Rent(2048);
+
+            // Act
+            var serializedResponse = updateResponseHeader.Serialize(memoryPool.Memory.Span);
+
+            // Assert
+            Assert.True(serializedResponse.ToArray().All(b => b == 0));
+        }
+
+        [Fact]
+        public void Serialize_WithFirmwareUpdates()
+        {
+            // setting up the twin in such a way that there are only firmware updates
+            var signatureBytes = Convert.FromBase64String("ABCD");
+            var keyCRC = 6789U;
+            var updateDataLength = 100U;
+            var updateResponseHeader = new CupsUpdateInfoResponseHeader
+            {
+                UpdateSignature = signatureBytes,
+                SignatureKeyCrc = keyCRC,
+                UpdateDataLength = updateDataLength,
+            };
+
+            using var memoryPool = MemoryPool<byte>.Shared.Rent(2048);
+
+            // Act
+            var serializedResponse = updateResponseHeader.Serialize(memoryPool.Memory.Span).GetReader();
+
+            // Assert
+            Assert.Equal(0, serializedResponse.Read()); //no uri
+            Assert.Equal(0, serializedResponse.Read()); //no uri
+            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); //no cred
+            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); //no cred
```

Comment:
Those `//no ...` comments are great, but would be good to align with distinct names used in the protocol documentation (here and in other tests in the same class):

```suggestion
            Assert.Equal(0, serializedResponse.Read()); // no cupsUri
            Assert.Equal(0, serializedResponse.Read()); // no tcUri
            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); // no cupsCred
            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); // no tcCred
```